continuum/transforms/custom.py

```diff
@@ -0,0 +1,88 @@
+from typing import Tuple, Union
+import numpy as np
+import torch
+from continuum.datasets import _ContinuumDataset
+
+
+class BackgroundSwap:
+    """Swap input image background with a randomly selected image from bg_images dataset
+
+    :param bg_images: background image dataset, must be normalized.
+    :param bg_label: label class from background image set
+    :param input_dim: input dimension of transform, excluding channels
+    """
+
+    def __init__(self, bg_images: _ContinuumDataset, input_dim: Tuple[int, int] = (28, 28),
+                 bg_label: int = None,
+                 normalize_bg: bool = True):
+        self.bg_label = bg_label
+        self.normalize_bg = normalize_bg
+        self.input_dim = input_dim
+
+        full = bg_images.get_data()
+
+        if bg_label is not None:
+            self.bg_images = full[0][np.where(full[1] == bg_label)]
+        else:
+            self.bg_images = full[0]
+
+    def _randcrop(self, img: Union[np.ndarray, torch.Tensor]) -> Union[np.ndarray, torch.Tensor]:
+        """Crop input image to self.input_dim shape
+        :param img: input image
+        """
+
+        crop_height = img.shape[0] >= self.input_dim[0]
+        crop_width = img.shape[1] >= self.input_dim[1]
+
+        x_crop = np.random.randint(0, img.shape[0] - self.input_dim[0])
+        y_crop = np.random.randint(0, img.shape[1] - self.input_dim[1])
+
+        if crop_width and crop_height:
+            return img[x_crop:x_crop + self.input_dim[0], y_crop: y_crop + self.input_dim[1], :]
+
+        if crop_width:
+            return img[:, y_crop: y_crop + self.input_dim[1], :]
+
+        if crop_height:
+            return img[x_crop:x_crop + self.input_dim[0], :, :]
+
+        raise Exception("Background image is smaller than foreground image")
+
+    def __call__(self, img: Union[np.ndarray, torch.Tensor],
+                 mask: Union[np.ndarray, torch.BoolTensor] = None
+                 ) -> Union[np.ndarray, torch.Tensor]:
+        """Splice input image foreground with randomly sampled background.
+
+        Inputting a torch.Tensor assumes the channel dim comes first,
+        while inputting a np.ndarray requires the channel dim to come second
+
+        :param img: input image, must be normalized
+        :param mask: boolean mask for the foreground of img, .5 threshold used by default
+        """
+
+        if isinstance(img, torch.Tensor):
+            img = img.repeat(3, 1, 1)
+            img = img.permute(1, 2, 0)
+
+        elif isinstance(img, np.ndarray):
+            img = np.expand_dims(img, 2)
+            img = np.concatenate([img, img, img], axis=2)
+
+        else:
+            raise NotImplementedError(f"Input type {type(img)} not implemented")
+
+        new_background = self.bg_images[np.random.randint(0, len(self.bg_images))]
+
+        if self.normalize_bg:
+            # TODO: don't hardcode normalization
```

Comment:
Propose here a lambda instead so it can be customize in input of the call method? or the init method?

Reply:
Do you mean lambda for the background normalization or for the background selection on line 74?